### PR TITLE
#407 building endpoint auth

### LINKED ISSUE #407: Add authentication to all the building endpoints
References user story #16

Add checking for an authentication token before running any building endpoints. If the token is invalid or doesn't exist, throw a 401.
If token is valid continue on with the request

In addition update the last modified user value to use the username in the claims object that is gotten from the auth token.

### PR DESCRIPTION
**Pull request by Kenny-Miller**
Closes #407 

Adds authentication to the front of all the building functions as well as updates the last modified user to use the username in the claims token for the add and update building endpoint.

### CODE REVIEW

File: BEIMA.Backend.FT/BuildingFT.cs [hunk 1]
```diff
@@ -66,6 +79,25 @@ await TestClient.GetBuilding(id)
...
+        {
+            var ex = Assert.ThrowsAsync<BeimaException>(async () =>
+                await UnauthorizedTestClient.GetBuilding("1234567890abcdef12345678")
+            );
+            Assert.IsNotNull(ex);
+            Assert.That(ex?.Message, Does.Contain("Invalid credentials."));
+            Assert.That(ex?.StatusCode, Is.EqualTo(HttpStatusCode.Unauthorized));
+        }
+
+        public void NoBuildingsInDatabase_Unauthorized_BuildingGet_EmptyId_ReturnsNotFound()
```
Comment (josephmoore99) [inline BEIMA.Backend.FT/BuildingFT.cs:92]: We already have a test case for this on NoBuildingsInDatabase_BuildingGet_ReturnsNotFound

**Comment by josephmoore99** [comment]
We should add unit tests specifically testing the unauthorized response, as I think we are doing this for all the other endpoints as well.

File: BEIMA.Backend.FT/BuildingFT.cs (hunk 1 above)
Comment (Kenny-Miller) [reply BEIMA.Backend.FT/BuildingFT.cs:92]: That test case doesn't test if the client is unauthorized though.

Comment (josephmoore99) [reply BEIMA.Backend.FT/BuildingFT.cs:92]: That's true, but this one still seems to be a not found error over an unauthorized error. But I suppose it shouldn't hurt to leave it in.

File: BEIMA.Backend.FT/BuildingFT.cs [hunk 2]
```diff
@@ -66,6 +79,25 @@ await TestClient.GetBuilding(id)
             Assert.That(ex?.StatusCode, Is.EqualTo(HttpStatusCode.NotFound));
         }
 
+        public void NoBuildingsInDatabase_Unauthorized_BuildingGet_ReturnsUnauthorized(string id)
```
Comment (josephmoore99) [inline BEIMA.Backend.FT/BuildingFT.cs:82]: This test is not being run because there is no [Test] tag.

**Review by josephmoore99 — APPROVED** [review]

**Review by thessbsu — APPROVED** [review]

**Review by roguedeleter — APPROVED** [review]

**Review by AshlynAdamson — APPROVED** [review]

**Pull request merged**